Unable to run tests because of broken dependency

When trying to run the tests locally, the command fails with the following error:

```
ImportError while loading conftest '/home/mjarosie/source/mjarosie/arsenic/tests/conftest.py'.
tests/conftest.py:13: in <module>
    from .app import build_app
tests/app.py:4: in <module>
    from jinja2 import FileSystemLoader, Environment
../../../.cache/pypoetry/virtualenvs/arsenic-AKF9VmY4-py3.10/lib/python3.10/site-packages/jinja2/__init__.py:12: in <module>
    from .environment import Environment
../../../.cache/pypoetry/virtualenvs/arsenic-AKF9VmY4-py3.10/lib/python3.10/site-packages/jinja2/environment.py:25: in <module>
    from .defaults import BLOCK_END_STRING
../../../.cache/pypoetry/virtualenvs/arsenic-AKF9VmY4-py3.10/lib/python3.10/site-packages/jinja2/defaults.py:3: in <module>
    from .filters import FILTERS as DEFAULT_FILTERS  # noqa: F401
../../../.cache/pypoetry/virtualenvs/arsenic-AKF9VmY4-py3.10/lib/python3.10/site-packages/jinja2/filters.py:13: in <module>
    from markupsafe import soft_unicode
E   ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/home/mjarosie/.cache/pypoetry/virtualenvs/arsenic-AKF9VmY4-py3.10/lib/python3.10/site-packages/markupsafe/__init__.py)
```

It's caused by using an old version of `Jinja2` with its broken dependency (`markupsafe`): https://github.com/pallets/markupsafe/issues/282

Steps to reproduce:

```bash
poetry install
poetry run pytest --verbose tests
```

One solution would be to pin `markupsafe` dependency to `2.0.1` as suggested in [this SO answer](https://stackoverflow.com/a/72747002). Another solution would be to update `Jinja2` dependency, which seems safe as it's a dev-dependency anyway.

EDIT:

It turns out the CI check is failing because of yet another outdated dependency. Invoking `black` causes the following error:

```
ImportError: cannot import name '_unicodefun' from 'click' 
```

More details about the cause: https://github.com/psf/black/issues/2964

EDIT:

After updating `Jinja2` to its latest version (`^3.1.2`) it turns out `sphinx-build` isn't happy because:

```
ImportError: cannot import name 'environmentfilter' from 'jinja2' 
```

Pinning to `Jinja2<3.1` [apparently should fix it](https://github.com/sphinx-doc/sphinx/issues/10291#issuecomment-1078046986).

EDIT:

Pinning to `Jinja2<3.1` did fix the issue and the build is green now. I've tried [upgrading](https://github.com/HENNGE/arsenic/pull/154/commits/575197c11ebf843ebc744c572eae653ecd4f151b) `sphinx` version instead which also results in a green build. As a side effect https://arsenic.readthedocs.io will get a refreshed look, such as a navigation bar which I was going to mention in another GitHub issue anyway.

### Old look
![image](https://user-images.githubusercontent.com/9082353/194726092-e8986723-3bf2-47ee-8f24-b0e4a30caadd.png)

### New look
![image](https://user-images.githubusercontent.com/9082353/194726098-f90a8ae5-9ccd-44ff-a4dd-0c64e1a78caf.png)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ structlog = "^20.1.0"
 aiohttp = ">=3"
 
 [tool.poetry.dev-dependencies]
-jinja2 = "^3.1.2"
+jinja2 = "<3.1"
 pytest = "^6.0.1"
 pytest-asyncio = "^0.14.0"
 sphinx = "^3.2.1"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ structlog = "^20.1.0"
 aiohttp = ">=3"
 
 [tool.poetry.dev-dependencies]
-jinja2 = "^2.11.2"
+jinja2 = "^3.1.2"
 pytest = "^6.0.1"
 pytest-asyncio = "^0.14.0"
 sphinx = "^3.2.1"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,7 +37,7 @@ sphinx = "^3.2.1"
 pytest-cov = "^2.10.1"
 asyncio-extras = "^1.3.2"
 pre-commit = "^2.7.1"
-black = "^20.8b1"
+black = "^22.10.0"
 Pillow = "^8.1.0"
 
 [tool.poetry.scripts]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,10 +30,10 @@ structlog = "^20.1.0"
 aiohttp = ">=3"
 
 [tool.poetry.dev-dependencies]
-jinja2 = "<3.1"
+jinja2 = "^3.1.2"
 pytest = "^6.0.1"
 pytest-asyncio = "^0.14.0"
-sphinx = "^3.2.1"
+sphinx = "^5.2.3"
 pytest-cov = "^2.10.1"
 asyncio-extras = "^1.3.2"
 pre-commit = "^2.7.1"

--- a/src/arsenic/helpers.py
+++ b/src/arsenic/helpers.py
@@ -8,7 +8,6 @@ if sys.platform != "win32":
     def configure_ie11_environment_cli():
         pass
 
-
 else:
     import ctypes
     import winreg

--- a/src/arsenic/services.py
+++ b/src/arsenic/services.py
@@ -52,7 +52,7 @@ async def subprocess_based_service(
                     ok = False
                 if ok:
                     return
-                await asyncio.sleep(start_timeout * 2 ** i)
+                await asyncio.sleep(start_timeout * 2**i)
 
         try:
             await asyncio.wait_for(wait_service(), timeout=start_timeout)

